### ISSUE TITLE
sqlite3.ProgrammingError: SQLite objects created in a thread can only be used in that same thread

### ISSUE THREAD

**Issue by skshetry** [iterative/dvc#4491]
### Please provide information about your setup

**Output of `dvc version`:**
It's 1.5.0. I don't have other information.

**Additional Information (if any):**
Reported on Discord: https://discord.com/channels/485586884165107732/563406153334128681/748445420191219742

```
Traceback (most recent call last):
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/main.py", line 54, in main
    ret = cmd.run()
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/command/imp.py", line 18, in run
    rev=self.args.rev,
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/imp.py", line 6, in imp
    return self.imp_url(path, out=out, erepo=erepo, frozen=True)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/__init__.py", line 34, in wrapper
    ret = f(repo, *args, **kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/scm_context.py", line 4, in run
    result = method(repo, *args, **kw)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/imp_url.py", line 54, in imp_url
    stage.run()
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/funcy/decorators.py", line 39, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/stage/decorators.py", line 35, in rwlocked
    return call()
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/funcy/decorators.py", line 60, in __call__
    return self._func(*self._args, **self._kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/stage/__init__.py", line 429, in run
    sync_import(self, dry, force)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/stage/imports.py", line 30, in sync_import
    stage.deps[0].download(stage.outs[0])
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/dependency/repo.py", line 95, in download
    _, _, cache_infos = repo.fetch_external([self.def_path])
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/external_repo.py", line 145, in fetch_external
    path, download_callback=download_update
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 297, in save_info
    return {self.PARAM_CHECKSUM: self.get_hash(path_info, **kwargs)}
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 266, in get_hash
    hash_ = self.get_dir_hash(path_info, **kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 282, in get_dir_hash
    dir_info = self._collect_dir(path_info, **kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 324, in _collect_dir
    new_hashes = self._calculate_hashes(not_in_state)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 309, in _calculate_hashes
    hashes = dict(zip(file_infos, tasks))
  File "/home/steve/.conda/envs/tf/lib/python3.6/concurrent/futures/_base.py", line 586, in result_iterator
    yield fs.pop().result()
  File "/home/steve/.conda/envs/tf/lib/python3.6/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/home/steve/.conda/envs/tf/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/home/steve/.conda/envs/tf/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/progress.py", line 126, in wrapped
    res = fn(*args, **kwargs)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/tree.py", line 418, in get_file_hash
    return self.dvctree.get_file_hash(path_info)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/tree.py", line 235, in get_file_hash
    return self._get_granular_checksum(path_info, out)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/repo/tree.py", line 49, in _get_granular_checksum
    dir_cache = out.get_dir_cache(remote=remote)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/output/base.py", line 384, in get_dir_cache
    if self.cache.changed_cache_file(self.checksum):
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/cache/base.py", line 315, in changed_cache_file
    actual = self.tree.get_hash(cache_info)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/tree/base.py", line 248, in get_hash
    hash_ = self.state.get(path_info)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/state.py", line 408, in get
    existing_record = self.get_state_record_for_inode(actual_inode)
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/state.py", line 352, in get_state_record_for_inode
    self._execute(cmd, (self._to_sqlite(inode),))
  File "/home/steve/.conda/envs/tf/lib/python3.6/site-packages/dvc/state.py", line 133, in _execute
    return self.cursor.execute(cmd, parameters)
sqlite3.ProgrammingError: SQLite objects created in a thread can only be used in that same thread. The object was created in thread id 139648838539008 and this is thread id 139648277903104.
------------------------------------------------------------
```

**Comment by skshetry** [comment]
Though I am unable to reproduce it in dir/file import/update/pull.

**Comment by efiop** [comment]
Should be fixed with diskcache in dvc 2.0
